compiler/src/dotty/tools/dotc/transform/SuperPhase.scala

```diff
@@ -0,0 +1,628 @@
+package dotty.tools
+package dotc
+package transform
+
+import core._
+import ast.Trees._
+import Contexts._, Phases._, Periods._, Symbols._, Decorators._
+import Flags.PackageVal
+import scala.annotation.tailrec
+import config.Printers.transforms
+import scala.util.control.NonFatal
+import reporting.trace
+
+/** A SuperPhase combines a number of mini-phases which are all executed in
+ *  a single tree traversal.
+ *
+ *  This is an evolution of the previous "TreeTransformers.scala", which was written by @DarkDimius and
+ *  is described in his thesis.
+ */
+object SuperPhase {
```

Comment:
Bikeshed: I'm not a big fan of the name `SuperPhase` since `super` already has a very specific meaning in Scala. Some possible alternatives: `MegaPhase`,`FusedPhase`, `JoinedPhase`.

Reply:
OK, let's go with MegaPhase. I like the sound of that :-)